game-core/src/main/java/games/strategy/triplea/delegate/ExecutionStack.java

```diff
@@ -28,37 +29,36 @@
  */
 public class ExecutionStack implements Serializable {
   private static final long serialVersionUID = -8675285470515074530L;
-  private IExecutable m_current;
-  @SuppressWarnings("JdkObsolete") // change to Deque/ArrayDeque upon next incompatible release
-  private final Stack<IExecutable> m_stack = new Stack<>();
+  private boolean isExecuting;
+  private final Deque<IExecutable> deque = new ArrayDeque<>();
 
   void execute(final IDelegateBridge bridge) {
-    // we were interrupted before, resume where we left off
-    if (m_current != null) {
-      m_stack.push(m_current);
+    isExecuting = true;
+    while (!deque.isEmpty()) {
+      final IExecutable executable = deque.pop();
+      try {
+        executable.execute(this, bridge);
+      } catch (final RuntimeException e) {
```

Comment:
Are you sure that a RuntimeException would always be thrown if this is interrupted?

Reply:
Well we know that it obviously can't be a checked exception.
So the only Throwables that would not be caught and rethrown here would be Errors that are usually thrown by the JVM (or AssertionErrors) and are not caught anywhere in the codebase AFAIK.
So any non-RuntimeException would instantly terminate the current thread and very likely the JVM along with it (for example in case of an OutOfMemoryError).
So I'm 99% sure we're safe here.